### PR TITLE
test: anchor unit test improvements

### PR DESCRIPTION
**Pull request by ilpepepig**
Closes DTT-293

### CODE REVIEW

File: test/unit/core/AnchorUnit.tree [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Anchor::constructor
+├── it should set registry
+└── it should set profileId
+
+Anchor::execute
+├── when caller is not the profileId owner
+│   └── it should revert
+├── when _target is the zero address
```
Comment (0xAustrian) [inline test/unit/core/AnchorUnit.tree]: Kind of a nitpick but we should avoid using `_` in when conditions according to the [docs](https://github.com/defi-wonderland/wonder-tests?tab=readme-ov-file#213-coverage): _There should be no underscore in the Given/When conditions (for clarity's sake once using Bulloak for scaffolding)_

**Review by 0xAustrian — APPROVED** [review]

File: test/unit/core/AnchorUnit.tree (hunk 1 above)
Comment (ilpepepig) [reply test/unit/core/AnchorUnit.tree]: fixed

**Review by 0xAustrian — APPROVED** [review]

File: test/unit/core/AnchorUnit.t.sol [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        vm.prank(profileOwner);
+
+        vm.expectRevert(Anchor.CALL_FAILED.selector); // Expect a revert with the CALL_FAILED error
+
+        _anchor.execute(address(0), _value, _data);
+    }
+
+    modifier whenCallerOwnsTheProfile() {
+        _;
+    }
```
Comment (0xOneTony) [inline test/unit/core/AnchorUnit.t.sol]: This is not used

File: test/unit/core/AnchorUnit.t.sol [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        (Anchor _anchor, address _registry, bytes32 _profileId) = _initAnchor();
+
+        address profileOwner = makeAddr("profileOwner");
+
+        vm.mockCall(
+            _registry,
+            abi.encodeWithSelector(Registry.isOwnerOfProfile.selector, _profileId, profileOwner),
+            abi.encode(true)
+        );
+        // vm.etch(targetAddr, code);
```
Comment (0xOneTony) [inline test/unit/core/AnchorUnit.t.sol]: remove this?

Comment (ilpepepig) [reply test/unit/core/AnchorUnit.t.sol]: Hmm I think it's needed

Comment (0xOneTony) [reply test/unit/core/AnchorUnit.t.sol]: why is it commented though?

Comment (ilpepepig) [reply test/unit/core/AnchorUnit.t.sol]: lol I'm blind. Removed!

**Review by 0xOneTony — APPROVED** [review]

**Review by 0xAustrian — APPROVED** [review]

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**